torch/csrc/jit/codegen/cuda/ops/alias.cpp

```diff
@@ -52,6 +52,33 @@ TensorView* applyViewTransforms(
 
 } // namespace
 
+TensorView* view(TensorView* x, DataType dtype) {
+  if (x->getDataType() == dtype) {
+    return x;
+  }
+  // TODO: support view(dtype) for dtypes of different size.
+  TORCH_INTERNAL_ASSERT(
+      dataTypeSize(x->getDataType().value()) == dataTypeSize(dtype));
+
+  std::vector<IterDomain*> out_domain;
+  // TODO: should the following line be:
+  //    TensorDomain::noReductions(x->getMaybeRFactorDomain());
+  // ? What does it mean?
+  auto inp_domain = x->domain()->getRootDomain();
+  out_domain.reserve(inp_domain.size());
+  for (auto d : inp_domain) {
+    out_domain.push_back(d->clone());
+  }
+  auto out = IrBuilder::create<TensorView>(
+      x->container(),
+      IrBuilder::create<TensorDomain>(
+          out_domain, std::vector<bool>(out_domain.size(), true)),
+      dtype);
```

Comment:
I am not sure if I am doing the correct thing here. The test passes, but I am not confident that the test will find all issues.

Reply:
Looks good except for the above `inp_domain`.